plugins/modules/network_compliance_workflow_manager.py

```diff
@@ -1173,108 +1169,276 @@ def get_sync_config_task_status(self, task_id, mgmt_ip_to_instance_id_map):
         }
 
+        # Check if all action_map keys are missing in self.want
+        if not any(action_param in self.want for action_param in action_map.keys()):
+            self.msg = "Network Compliance operation(s) are not required for the provided input parameters in the Cisco Catalyst Center."
+            self.set_operation_result("ok", False, self.msg, "INFO")
+            return self
+
+        final_status_list = []
+        result_details = {}
+
         # Iterate through the action map and execute specified actions
         for action_param, (action_func, status_func) in action_map.items():
-
+            req_action_param = self.want.get(action_param)
             # Execute the action and check its status
-            if self.want.get(action_param):
+            if req_action_param:
+                self.log("Executing action function: {0} with params: {1}".format(action_func.__name__, req_action_param), "INFO")
                 if action_param == "run_compliance_params":
-                    self.log("Performing '{0}' operation...".format(action_func.__name__), "DEBUG")
                     batch_size = self.want.get("run_compliance_batch_size")
                     result_task_id = action_func(self.want.get(action_param), batch_size=batch_size)
                 else:
-                    self.log("Performing '{0}' operation...".format(action_func.__name__), "DEBUG")
                     result_task_id = action_func(self.want.get(action_param))
 
                 if not result_task_id:
                     self.msg = "An error occurred while retrieving the task_id of the {0} operation.".format(action_func.__name__)
                     self.set_operation_result("failed", False, self.msg, "CRITICAL")
                 else:
+                    self.log("Task Id: {0} returned from the action function: {1}".format(result_task_id, action_func.__name__), "DEBUG")
                     status_func(result_task_id, self.want.get("mgmt_ip_to_instance_id_map")).check_return_status()
+                    result = self.msg
+                    result_details.update(result)
 
+        if config.get("sync_device_config"):
+            skipped_sync_device_configs_list = set(self.skipped_sync_device_configs_list)
+            if skipped_sync_device_configs_list:
+                result_details["Sync Device Configuration operation Skipped for following device(s)"] = {
+                    "skipped_count": len(skipped_sync_device_configs_list),
+                    "skipped_devices": skipped_sync_device_configs_list
+                }
+
+        skipped_run_compliance_devices_list = set(self.skipped_run_compliance_devices_list)
+        if skipped_run_compliance_devices_list:
+            result_details["Run Compliance Check Skipped for following device(s)"] = {
+                "skipped_count": len(skipped_run_compliance_devices_list),
+                "skipped_devices": skipped_run_compliance_devices_list
+            }
```

Comment:
self.log("Run Compliance Check skipped for devices: {0}".format(skipped_run_compliance_devices_list), "DEBUG")